tools/Makefile.shared

```diff
@@ -250,6 +250,30 @@ clean::
 
 beforedepend::
 
+# caml-types using cmt files
+
+CMT_LOOKUP= \
+            ../compilerlibs/ocamlcommon.cma \
+            ../compilerlibs/ocamlbytecomp.cma \
+            cmt_lookup.cmo
+
+CMT_LOOKUP_OPT1 = $(CMT_LOOKUP:.cmo=.cmx)
+CMT_LOOKUP_OPT = $(CMT_LOOKUP_OPT1:.cma=.cmxa)
+
+cmt_lookup: $(CMT_LOOKUP)
+	$(CAMLC) $(LINKFLAGS) -o cmt_lookup $(CMT_LOOKUP)
+
+cmt_lookup.opt: $(CMT_LOOKUP_OPT)
+	$(CAMLOPT) $(LINKFLAGS) -o cmt_lookup.opt $(CMT_LOOKUP_OPT)
+
+clean::
+	rm -f cmt_lookup cmt_lookup.opt
+
+beforedepend::
+
+install::
+	cp cmt_lookup cmt_lookup.opt $(INSTALL_BINDIR)/
```

Comment:
You should protect `$(INSTALL_BINDIR)/` against spaces — the build fails on Windows because of that.